static_topology/stackstate_checks/static_topology/static_topology.py

```diff
@@ -0,0 +1,136 @@
+# (C) StackState 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import csv
+import codecs
+
+from stackstate_checks.base import ConfigurationError, AgentCheck, TopologyInstance
+
+
+class StaticTopologyCheck(AgentCheck):
+    SERVICE_CHECK_NAME = "StaticTopology"
+    INSTANCE_TYPE = "StaticTopology"
+
+    def get_instance_key(self, instance):
+        if "components_file" not in instance:
+            raise ConfigurationError('Static topology instance missing "components_file" value.')
+
+        return TopologyInstance(self.INSTANCE_TYPE, instance["components_file"])
+
+    def check(self, instance):
+        if 'components_file' not in instance:
+            raise ConfigurationError('Static topology instance missing "components_file" value.')
+        if 'relations_file' not in instance:
+            raise ConfigurationError('Static topology instance missing "relations_file" value.')
+        if 'type' not in instance:
+            raise ConfigurationError('Static topology instance missing "type" value.')
+
+        instance_tags = instance['tags'] if 'tags' in instance else []
+
+        if instance['type'].lower() == "csv":
+            component_file = instance['components_file']
+            relation_file = instance['relations_file']
+            delimiter = instance['delimiter']
+            instance_tags.extend(["csv.component:%s" % component_file, "csv.relation:%s" % relation_file])
+            try:
+                self.start_snapshot()
+                self.handle_component_csv(component_file, delimiter, instance_tags)
+                if relation_file:
+                    self.handle_relation_csv(relation_file, delimiter, instance_tags)
+            except ConfigurationError as e:
+                self.log.error(str(e))
+                raise e
+            except Exception as e:
+                self.log.exception(str(e))
+                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=str(e), tags=instance_tags)
+            finally:
+                self.stop_snapshot()
+        else:
+            raise ConfigurationError('Static topology instance only supports type CSV.')
```

Comment:
we don't throw exception from the check but rather use the `service_check` critical

Reply:
this is removed and converted into `service_checks`.